src/category_theory/sites/cocontinuous.lean

```diff
@@ -0,0 +1,214 @@
+/-
+Copyright (c) 2021 Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang
+-/
+import category_theory.sites.sheaf
+import category_theory.limits.kan_extension
+
+/-!
+# Cocontinuous functors between sites.
+
+We define cocontinuous functors between sites as functors that pulls covering sieves back to
+covering sieves. This seems stronger that the definition found in stacks project,
+but they are actually equivalent via `category_theory.grothendieck_topology.superset_covering`
+(The precise statement not formalized, but follows from it quite trivially).
+
+This concept is also called the "covering lifting property" as defined in [MM92] Chapter VII,
+Section 10, and should not be confused with the general definition of cocontinuous functors
+between categories as functors preserving small colimits.
+
+## Main Results
+- `category_theory.sites.Ran_is_sheaf_of_cocontinuous`: If `u : C ⥤ D` is cocontinuous, then
+`Ran u.op` (`ₚu`) as a functor `(Cᵒᵖ ⥤ A) ⥤ (Dᵒᵖ ⥤ A)` of presheaves maps sheaves to sheaves.
+
+## References
+
+* [S. MacLane, I. Moerdijk, *Sheaves in Geometry and Logic*][MM92]
+* https://stacks.math.columbia.edu/tag/00XI
+
+-/
+
+universes u
+noncomputable theory
+
+open category_theory
+open opposite
+open category_theory.presieve.family_of_elements
+open category_theory.presieve
+open category_theory.limits
+
+namespace category_theory
+namespace sites
+variables {C D A : Type u} [category.{u} C] [category.{u} D] [category.{u} A] [has_limits A]
+variables {J : grothendieck_topology C} {K : grothendieck_topology D}
+
+/--
+A functor `u : (C, J) ⥤ (D, K)` between sites is called cocontinuous if for all covering sieves
+`R` in `D`, `R.pullback u` is a covering sieve in `C`.
+-/
+structure cocontinuous (J : grothendieck_topology C) (K : grothendieck_topology D) (u : C ⥤ D) :=
+(cover_lift : ∀ {U : C} {S : sieve (u.obj U)} (hS : S ∈ K (u.obj U)), S.functor_pullback u ∈ J U)
+
+/-- A trivial example to make the inhabited linter happy. -/
+instance id_cocontinuous : inhabited (cocontinuous J J (𝟭 _)) :=
+⟨⟨λ _ S h, by { convert h, simp }⟩⟩
+
```

Comment:
Could you please add a lemma `cocontinuous.comp` that proves that the composition of cocontinuous functors is cocontinuous?